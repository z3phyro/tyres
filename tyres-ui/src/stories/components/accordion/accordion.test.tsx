import { test, expect } from "vitest";
import { render } from "@solidjs/testing-library";
import Accordion from "./accordion";
import { BackgroundColor, EUiVariant } from "~/core/types/ui-variants.type";

/*Commented until issue with jsx gets solved*/
test("Accordion renders", () => {
  const { getByRole } = render(() => <Accordion items={[]} />);

  const accordion = getByRole("list") as HTMLUListElement;
  expect(accordion).toBeTruthy();
});

test("Accordion renders with custom items", () => {
  const { getByRole } = render(() => (
    <Accordion
      items={[
        {
          header: "Header 1",
          content: "Content 1",
          key: "1",
        },
        {
          header: "Header 2",
          content: "Content 2",
          key: "2",
        },
      ]}
    />
  ));

  const accordion = getByRole("list");
  expect(accordion.children.length).toBe(2);
});

test("Accordion renders with default value", () => {
  const { getByRole } = render(() => (
    <Accordion
      items={[
        {
          header: "Header 1",
          content: "Content 1",
          key: "1",
        },
        {
          header: "Header 2",
          content: "Content 2",
          key: "2",
        },
      ]}
      defaultValue={["1"]}
    />
  ));

  const accordion = getByRole("list");
  expect(accordion.children.length).toBe(2);
  expect(accordion.children[0].getElementsByTagName("button")[0].getAttribute("aria-expanded")).toBe("true");
  expect(accordion.children[1].getElementsByTagName("button")[0].getAttribute("aria-expanded")).toBe("false");
});

// Check accordion items variants

test("Accordion renders with success variant", () => {
  const { getByRole } = render(() => (
    <Accordion
      variant={EUiVariant.Success}
      items={[
        {
          header: "Header 1",
          content: "Content 1",
          key: "1",
        },
      ]}
    />
  ));

  const accordion = getByRole("list");
  expect(accordion.children.length).toBe(1);
  expect(accordion.children[0].getElementsByTagName("h3")[0]).toHaveClass(BackgroundColor[EUiVariant.Success]);
});

test("Accordion has correct header", () => {
  const { getByRole } = render(() => (
    <Accordion
      items={[
        {
          header: "Header 1",
          content: "Content 1",
          key: "1",
        },
      ]}
    />
  ));

  const accordion = getByRole("list");
  expect(accordion.children.length).toBe(1);
  expect(accordion.children[0].getElementsByTagName("button")[0]).toHaveTextContent("Header 1");
});

test("Accordion snapshot test", () => {
  const { asFragment } = render(() => (
    <Accordion
      items={[
        {
          header: "Header 1",
          content: "Content 1",
          key: "1",
        },
        {
          header: "Header 2",
          content: "Content 2",
          key: "2",
        },
      ]}
    />
  ));

  expect(asFragment()).toMatchSnapshot();
});
