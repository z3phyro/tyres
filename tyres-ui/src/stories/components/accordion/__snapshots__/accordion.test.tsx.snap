// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion snapshot test 1`] = `"<div id="accordion-cl-17" class="accordion rounded" role="list"><div data-closed="" id="accordion-cl-17-item-cl-18" class="accordion__item "><h3 data-closed="" class="accordion__item-header rounded border-gray-200 bg-white text-black"><button type="button" aria-expanded="false" data-closed="" data-key="1" id="accordion-cl-17-item-cl-18-trigger" class="accordion__item-trigger">Header 1<svg class="accordion__item-trigger-icon" aria-hidden="true" fill="none" stroke-width="2" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" viewBox="0 0 24 24" height="1em" width="1em" style="overflow: visible; color: currentcolor;"><path d="M6 9 12 15 18 9"></path></svg></button></h3></div><div data-closed="" id="accordion-cl-17-item-cl-20" class="accordion__item "><h3 data-closed="" class="accordion__item-header rounded border-gray-200 bg-white text-black"><button type="button" aria-expanded="false" data-closed="" data-key="2" id="accordion-cl-17-item-cl-20-trigger" class="accordion__item-trigger">Header 2<svg class="accordion__item-trigger-icon" aria-hidden="true" fill="none" stroke-width="2" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" viewBox="0 0 24 24" height="1em" width="1em" style="overflow: visible; color: currentcolor;"><path d="M6 9 12 15 18 9"></path></svg></button></h3></div></div>"`;
